sdk/src/main/java/com/hedera/hashgraph/sdk/Client.java

```diff
@@ -404,6 +401,25 @@ private void cancelScheduledNetworkUpdate() {
         }
     }
 
+    private void cancelAllSubscriptions() {
+        ArrayList<SubscriptionHandle> handles = new ArrayList<>(subscriptions);
+        for (SubscriptionHandle handle : handles) {
+            handle.unsubscribe();
+        }
```

Comment:
No need for a copy. This is handled within a synchronized close.

```suggestion
    private void cancelAllSubscriptions() {
        subscriptions.forEach(SubscriptionHandle::unsubscribe);
```

Reply:
The copy is needed to avoid ConcurrentModificationException after the first item is removed from the set/collection (that happens in onUnsubscribe - client.untrackSubscription(subscriptionHandle)).

Reply:
Think we should use `ConcurrentHashMap.newKeySet()` then we can avoid this copy.